UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs

```diff
@@ -147,7 +151,33 @@ public async Task<string> CreateLargeExchangeSet(SalesCatalogueServiceResponseQu
             await Task.WhenAll(ParallelCreateFolderTaskForCatlogFile);
             ParallelCreateFolderTaskForCatlogFile.Clear();
 
-            return "Large Exchange Set Created Successfully";
+            List<Task<bool>> parallelZipUploadTasks = new List<Task<bool>> { };
+            Parallel.ForEach(dvdNumbers, dvdNumber =>
+            {
+                var largeMediaExchangeSetPath = Path.Combine(homeDirectoryPath, currentUtcDate, message.BatchId, string.Format(largeExchangeSetFolderName, dvdNumber.ToString()));
+
+                parallelZipUploadTasks.Add(PackageAndUploadLargeMediaExchangeSetZipFileToFileShareService(message.BatchId, largeMediaExchangeSetPath, largeMediaExchangeSetZipFilePath, message.CorrelationId, string.Format(largeExchangeSetFolderName, dvdNumber.ToString())));
+            });
+
+            await Task.WhenAll(parallelZipUploadTasks);
+            bool isZipAndUploadSuccessful = await Task.FromResult(parallelZipUploadTasks.All(x => x.Result.Equals(true)));
+
+            if (isZipAndUploadSuccessful)
+            {
+                parallelZipUploadTasks.Clear();
+                var isBatchCommitted = await fulfilmentFileShareService.CommitLargeMediaExchangeSet(message.BatchId, largeMediaExchangeSetZipFilePath, message.CorrelationId);
+                createExchangeSetTaskCompletedAt = DateTime.UtcNow;
+                if (isBatchCommitted)
+                {
+                    logger.LogInformation(EventIds.ExchangeSetCreated.ToEventId(), "Large media exchange set is created for BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId}", message.BatchId, message.CorrelationId);
+                    monitorHelper.MonitorRequest("Create Exchange Set Task", createExchangeSetTaskStartedAt, createExchangeSetTaskCompletedAt, message.CorrelationId, null, null, null, message.BatchId);
+                    return "Large Media Exchange Set Created Successfully";
+                }
+            }
+
+            createExchangeSetTaskCompletedAt = DateTime.UtcNow;
```

Comment:
Just pass DateTime.UtcNow in the parameter rather than use a dedicated variable.

Reply:
Resolved